docs/design/l2-peer-protocol-spec.md

```diff
@@ -0,0 +1,507 @@
+---
+title: CL L2 Peer Protocol Spec
+author:
+  - "@waalge"
+  - "@paluh"
+---
+
+## Intro
+
+### Scope
+
+> Key points:
+>
+> - We will use libp2p so `ping` will be provided out of the box and we don't
+>   cover that part.
+> - Transport layer with encryption will be covered by separate spec but we will
+>   relay on standard libp2p capabilities.
+>   - We assume that transport layer is reliable and delivers messages in order.
+
+This document covers the communication protocol between two peers - CL nodes
+which wish to establish and manage lightning channels between each other. It
+defines messages and exchange flows required to achieve this task. We loosely
+follow and cover the scope of the BLN
+[bolt#0](https://github.com/Lighting/bolts/blob/master/00-introduction.md)
+[bolt#1](https://github.com/Lighting/bolts/blob/master/01-messaging.md) and
+[bolt#2](https://github.com/lightning/bolts/blob/master/02-peer-protocol.md) in
+this document. Please note that we miss interactive transaction building because
```

Comment:
I think "non-aims" is a valid section in its own right. 

My perspective is that batching one's own channel stepping is way more enabling than coordinating tx building with a partner. I don't think we need to be apologetic about this. 
Maybe I'd reconsider wen `tx_fee` > 2$.